README.md

```diff
@@ -3,16 +3,227 @@
+every resource that has a kind listed in `possibleGVKs` and looks for ownership from a specific
+resource. This helps guard against deleting resources that are not created by the code in question.
+The list is iterated through and any resource which is **not** in the cache will be deleted. This
+works because a reconciliation should ensure that **every** resource they care about is
+written/updated.
+
+Objects which have a k8s kind in the `protectedGVK` list will not be deleted by the Resource Cache.
+
+### Optimisations
+Certain optimisations are present to speed things up and help with optimisation.
+
+* When an item is retrieved from the cache, during the `Create()` call, a copy is stored. When
+  alterations to the object are made through `Update()`, these are made to a new copy of the
+  resource. When an apply is made, the Resource Cache compares the *initial version* to the *updated
+  version*. If there is no change, then nothing is written to k8s and hence API calls are saved.
+  This does come an some expense to the Resource Cache's resources, but in practice it is often far
+  more beneficial to spend cycles checking this than making thousands of unnecessary writes to k8s.
+
+* The Resource Cache sets a flag on the object inside the cache when the `Create()` is used so that
+  it is aware if the object needs to be created or updated. Usually working with the k8s client
+  requires the developer to know this or check for it, the Resource Cache makes this step
+  unnecessary.
+
+* The `Reconcile()` function can be costly as it needs to list every object of every kind in the
+  `possibleGVKs` list. To minimise this, options can be passed to Reconcile to allow it to filter
+  objects. This is often used to filter to within a certain namespace, or a certain label.
+
+* Certain objects are always written before others, or rather certain objects are always written
+  last. `Deployments/Jobs` are always written at the end of the `ApplyAll()` because they often rely
+  on other objects to be written out. Particularly if the reconciliation is happening for the first
+  time, these dependent objects may not exist and will cause a deployment to spin up pods which will
+  fail with a `ConfigCreateError`. Ordering resources like this allows for the best chance of
+  successful pod deployment.
+
+### WriteNow support
+There are some situations where a resource requires to be written immediately and not wait for an
+`ApplyAll()`. In this case, it will be skipped in the `ApplyAll()` step as an `Update()` call will
+immediately update it in k8s. This should be used with care. The object should not be updated after
+the initial update, though there is nothing to prevent this from happening, it ill simply result in
+a second API write.
+
+```go
+NewSingleResourceIdent("prov", "purpose", &core.ConfigMap{}, rc.ResourceOptions{WriteNow: true})
+```
+
+### possibleGVKs and strict mode
```

Comment:
Should this be moved up to the Create part because it is germane to creation?

Reply:
yeh I think so